Update readme file

vitbaq opened the pull request:
Closes #23 
Depends on #20 
Changes:
 - Written in reStructuredText
 - Update system requirements
 - Organize section "How to Build it"
 - Add result of the build
 - Update How to write the SD Card
 - Add section how to monitoring the gateway

This patch also create a new file HACKING.rst with information to
use Vagrant and different ways to build.

pauloserrafh commented on HACKING.rst:
```diff
@@ -0,0 +1,112 @@
+
+.. code-block:: bash
+
+   $ make O=/path/to/my/output/folder BR2_GATEWAY_HARDWARE=configs/<myraspberrypi_model>_defconfig knot_gateway
```
The process to build changed. No need to use `BR2_GATEWAY_HARDWARE` anymore. Also, the defconfigs is now `knot_raspberrypi3_defconfig` (and `knot_raspberrypi0w` to be include with #20).

pauloserrafh commented on HACKING.rst:
```diff
@@ -0,0 +1,112 @@
+.. code-block:: bash
+
+   $ make O=/path/to/my/output/folder BR2_GATEWAY_HARDWARE=configs/<myraspberrypi_model>_defconfig knot_gateway
+   $ make O=/path/to/my/output/folder menuconfig
```
menuconfig is an optional command, this should be noted.

pauloserrafh commented on HACKING.rst:
```diff
@@ -0,0 +1,112 @@
+
+.. code-block:: bash
+
+   $ make O=../output BR2_GATEWAY_HARDWARE=configs/<myraspberrypi_model>_defconfig knot_gateway
```
Same as above

pauloserrafh commented on HACKING.rst:
```diff
@@ -0,0 +1,112 @@
+.. code-block:: bash
+
+   $ make O=../output BR2_GATEWAY_HARDWARE=configs/<myraspberrypi_model>_defconfig knot_gateway
+   $ make O=../output menuconfig
```
Same as above

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+
+   .. code-block:: bash
+
+      sudo apt-get install build-essential automake \
```
apt-get is the package manager for Ubuntu/Debian. Other distros have different managers, add a note or something like that.

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+The knot_<board>_defconfig configuration is a minimal configuration with all
+that is required to bring the KNoT platform on RaspberryPi.
+
+**Warning**: You will need to have access to the network, since Buildroot will
```
Refer to HACKING.rst for an offline build.

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+in the output/images/ directory.
+
+It is possible to erase and write the SD Card using native OS application to
+manage and configure disk drives. Similarly to **Disks** on Ubuntu.
```
I understand that the tutorial is for linux-based systems, but it might be good to point out that once you have the `.img` file you could copy/move to any other operating system and use whatever flasher you want.

Perhaps it would be good to refer to a well known (and multi-platform) image flasher. such as [etcher](https://www.balena.io/etcher/).

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+
+      sudo dd if=output/images/sdcard.img of=/dev/sdX
+
+   **Tip**: See ``dd`` manual for more options. Like to see the progress you can add ``status=progress`` in the end of above ``dd`` command.
```
Add `status=progress`to the command line.  (Remember our user experience, most likely they'll copy and paste the command and not read the note.
Change **Tip** to:
**Note**: ``status=progress`` is an optional argument to show the progress of the command execution. See ``dd`` manual for more options.

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+
+   arp-scan -l | grep b8:27:eb
+
+**Note**: It is possible needed to install ``arp-scan``. To do this running
```
There are some concordance issues here.  Also, `apt-get` is for Ubuntu/Debian.

suggestion:
**Note**: It is necessary to have ``arp-scan`` package installed. If you're running a Ubuntu/Debian based distro, run ``sudo apt-get install arp-scan`` on terminal.

pauloserrafh commented on README.rst:
```diff
@@ -0,0 +1,171 @@
+This commands will return the IP address the router have assigned to the
+KNot gateway.
+
+To access the gateway using ``ssh``, run the followed command in a linux
```
[...]run the **following** command[...]
(perhaps **the comamnd below** is even better)

pauloserrafh commented on HACKING.rst:
```diff
@@ -0,0 +1,112 @@
+KNoT Network of Things - Gateway - HACKING
+==========================================
+
+Use a Vagrant box (experimental)
```
Is this vagrant box functional? I think it's (very) outdated. Perhaps add a note about it to be removed latter after testing.

vitbaq replied:
The vagrant box is functional, a member of the team has used already. But @joaoaneto will test the steps to make sure that is correct, a new issue will be created.

pauloserrafh reviewed (changes requested):
This patch could be broken in two or the commit message should be rewritten (or both).

vitbaq commented:
> This patch could be broken in two or the commit message should be rewritten (or both).

Can you be more specific? In which parts should I break it?

I force-pushed a change breaking the commit and rewriting the commit message.

vitbaq committed:
readme: Change readme to use reStructuredText

Signed-off-by: Vitor Barros <vba@cesar.org.br>

vitbaq committed:
readme: Update readme content

This patch has the following changes:
- Written in reStructuredTest syntax.
- Update system requirements
- Organize section about Vagrant and Builds
- Update result of the build
- Update How to write the SD Card
- Add section of how to monitoring the gateway

Signed-off-by: Vitor Barros <vba@cesar.org.br>

vitbaq committed:
readme: Create HACKING file

This patch create a new file called HACKING with extra informations
moved from README file.

Signed-off-by: Vitor Barros <vba@cesar.org.br>

pauloserrafh commented on README.rst:
```diff
@@ -1,297 +1,339 @@
-Copyright (C) 1999-2005 by Erik Andersen <andersen@codepoet.org>  
-Copyright (C) 2006-2014 by the Buildroot developers <buildroot@uclibc.org>  
-Copyright (C) 2014 by the Buildroot developers <buildroot@buildroot.org>  
-Copyright (C) 2016 by the C.E.S.A.R KNoT developers <knot-devel@cesar.org.br>
```
Can/should we remove this Copyright notices?

vitbaq replied:
On the mainline of Buildroot, this disclaimer does not exist.
The same disclaimer can be found on the Makefile and the COPYING file has the all license stuff.
So, in my opinion, we can remove this disclaimer. What do you think? Maybe we can leave the disclaimer because of CESAR Copyright.

pauloserrafh replied:
I agree that since BR removed from their README, it's OK. But need confirmation on the CESAR.

vitbaq replied:
I talked with Tiago Barros and he finds OK without this disclaimer.

pauloserrafh reviewed (changes requested):
Set as request changes just until we have defined the disclaimer question.
